environment_internal_test.go

```diff
@@ -0,0 +1,367 @@
+package gantry
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"testing"
+	"text/template"
+
+	"github.com/ad-freiburg/gantry/types"
+)
+
+func TestPipelineEnvironmentCreatedTemplateParserDirectSubstitution(t *testing.T) {
+	cases := []struct {
+		in            string
+		out           string
+		err           bool
+		substitutions types.MappingWithEquals
```

Comment:
A bit OT but I think the `MappingWithEquals` type could have a more useful name. To me it looks like this is just a string to string mapping for JSON/YAML. So maybe just call it `StringMap` which would also match `types.StringSet`

Reply:
`StringMap` sounds nice, I'll try that and make a new PR.